### DISCUSSION TITLE
Serve pyscript with Access-Control-Allow-Origin header

### DISCUSSION THREAD

**Discussion by joemarshall** [pyscript/pyscript#1199] [Proposals]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other feature requests and couldn't find a duplicate (including also the ``type-feature`` tag)
- [X] I confirmed that it's not related to another project area (see the above section)

### What is the idea?

Add Access-Control-Allow-Origin:* to the response headers for pyscript.js / .css on the default pyscript.net server.

Or serve it on a CDN that has CORS headers set.

### Why is this needed

If your site is cross-origin isolated (e.g. if you're using something which requires SharedArrayBuffer and Atomics), you currently can't use pyscript without installing it on your local server.

**Comment by pauleveritt** [comment]
OT: Would love to see anything you have @joemarshall about Atomics.

**Comment by joemarshall** [comment]
@pauleveritt 

Check this out - pyodide based streaming http requests (for progress or for consuming data as it comes in) implemented using atomics and SharedArrayBuffer

https://github.com/joemarshall/pyodide-http

Annoyingly Atomics.wait doesn't work on the UI thread, so that particular example still won't work in pyscript without adding horrible spin-waiting things in place of the atomics.wait, but it demonstrates the basic use of the atomics stuff.

**Comment by pauleveritt** [comment]
@joemarshall Can I email you about the implementation?

I did a PoC last weekend of PyScript with the custom elements completely implemented in Python. I used a messaging pattern, UI<->worker<->Pyodide. What I really wanted was something like Redux. Tried two-way generators but...I'm not good enough at that. 😇

If you don't mind, I'd like to pick your brain and learn.

**Comment by Hoai-Phong** [comment]
@joemarshall  @jezdez  @mattpap 
 How to grant access-Control-Allow-Origin permission. I am facing still not resolved.
![image](https://github.com/user-attachments/assets/6b997373-69ad-48c0-8243-7f82a3ed2688)
